core/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -180,7 +180,7 @@ protected XContentBuilder newXContentBuilder(XContentType type, OutputStream str
      * the state.
      */
     public final T read(Path file) throws IOException {
-        try (Directory dir = newDirectory(file.getParent())) {
+        try (Directory dir = newReadOnlyDirectory(file.getParent())) {
```

Comment:
can't we just check if the directory exists before we open that directory and if it doesn't exist we don't open it? would that be the simplest fix?